dotcom-rendering/src/components/SignInGate/dismissGate.ts

```diff
@@ -1,135 +1,113 @@
-	}
+	const prefs = getSigninGatePrefsSafely();
+	prefs[localStorageDismissedDateKey(variant, name)] =
+		new Date().toISOString();
+	setSigninGatePrefs(prefs);
 };
 
 export const unsetUserDismissedGate = (variant: string, name: string): void => {
-	try {
-		const prefs = getSigninGatePrefsSafely();
-		delete prefs[localStorageDismissedDateKey(variant, name)];
-		setSigninGatePrefs(prefs);
-	} catch (error) {
-		// Alas, sometimes localstorage isn't available
-	}
+	const prefs = getSigninGatePrefsSafely();
+	delete prefs[localStorageDismissedDateKey(variant, name)];
+	setSigninGatePrefs(prefs);
 };
 
-// Check if the user has dismissed the gate by checking the user preferences,
-// name is optional, but can be used to differentiate between multiple sign in gate tests
-//
-// This is set in local storage with the following shape:
-//
-// key:   gu.prefs.sign-in-gate
-// value: {"testVariantName":"2020-07-01T10:55:09.085Z"}
-//
-// We extract the value using the key, which remains constant
-// and the from within the value object we look up the variant we are looking for
+/**
+ * Check if the user has dismissed the gate by checking the user preferences,
+ * name is optional, but can be used to differentiate between multiple sign in gate tests
+ *
+ * This is set in local storage with the following shape:
+ *
+ * key:   gu.prefs.sign-in-gate
+ * value: {"testVariantName":"2020-07-01T10:55:09.085Z"}
+ *
+ * We extract the value using the key, which remains constant
+ * and the from within the value object we look up the variant we are looking for
+ */
 export const hasUserDismissedGate = (
 	variant: string,
 	name: string,
-	window?: number, // represents hours - only use if the gate should reshow after X hrs (dismissal window)
+	dismissalWindow?: number, // represents hours - only use if the gate should reshow after X hrs (dismissal window)
 ): boolean => {
-	try {
-		const prefs = getSigninGatePrefsSafely();
-		// checks if a dismissal occurred within a given window timeframe in hours
-		if (window !== undefined) {
-			// checks if prefs is empty, ie. the user has not dismissed gate before.
-			if (!prefs[localStorageDismissedDateKey(variant, name)]) {
-				return false;
-			}
-
-			const dismissalTZ = Date.parse(
-				prefs[localStorageDismissedDateKey(variant, name)],
-			);
-			const hours = (Date.now() - dismissalTZ) / 36e5; //  36e5 is the scientific notation for 60*60*1000, which converts the milliseconds difference into hours.
-
-			if (hours >= window) {
-				unsetUserDismissedGate(variant, name);
-				return false;
-			}
-			return true;
+	const prefs = getSigninGatePrefsSafely();
+	const pref = prefs[localStorageDismissedDateKey(variant, name)];
+	// checks if a dismissal occurred within a given window timeframe in hours
+	if (dismissalWindow !== undefined) {
+		// checks if prefs is empty, ie. the user has not dismissed gate before.
+		if (isUndefined(pref) || typeof pref === 'number') {
```

Comment:
This might be simpler:

```suggestion
		if (!isString(pref)) {
```